datajunction-server/datajunction_server/internal/materializations.py

```diff
@@ -214,6 +215,12 @@ def create_new_materialization(
     if current_revision.type == NodeType.CUBE:
         # Check to see if a default materialization was already configured, so that we
         # can copy over the default cube setup and layer on specific config as needed
+        if not temporal_partition:
+            raise DJInvalidInputException(
+                "The cube materialization cannot be configured if there is no "
+                "temporal partition specified on the cube. Please set at least one cube"
```

Comment:
Maybe make it a little clearer by `... Please make sure at least one cube element has a temporal partition defined.`

Reply:
Good point, changing this